test/remote/gateways/remote_reach_test.rb

```diff
@@ -0,0 +1,134 @@
+require 'test_helper'
+
+class RemoteReachTest < Test::Unit::TestCase
+  def setup
+    @gateway = ReachGateway.new(fixtures(:reach))
+    @amount = 100
+    @credit_card = credit_card('4444333322221111', {
+      month: 3,
+      year: 2030,
+      verification_value: 737
+    })
+    @declined_card = credit_card('4000300011112220')
+    @options = {
+      email: 'johndoe@reach.com',
+      order_id: '123',
+      description: 'Store Purchase',
+      currency: 'USD',
+      billing_address: {
+        address1: '1670',
+        address2: '1670 NW 82ND AVE',
+        city: 'Miami',
+        state: 'FL',
+        zip: '32191',
+        country: 'US'
+      }
+    }
+  end
+
+  def test_successful_authorize
+    response = @gateway.authorize(@amount, @credit_card, @options)
+
+    assert_success response
+    assert response.params['response'][:Authorized]
+    assert response.params['response'][:OrderId]
+  end
+
+  def test_failed_authorize
+    @options[:currency] = 'PPP'
+    @options.delete(:email)
+    response = @gateway.authorize(@amount, @credit_card, @options)
+
+    assert_failure response
+    assert_equal 'Invalid ConsumerCurrency', response.message
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card, @options)
+
+    assert_success response
+    assert response.params['response'][:Authorized]
+    assert response.params['response'][:OrderId]
+  end
+
+  def test_failed_purchase
+    response = @gateway.purchase(@amount, @declined_card, @options)
+    assert_failure response
+    assert_equal 'NotATestCard', response.message
+  end
+
+  def test_successful_purchase_with_fingerprint
+    @options[:fingerprint] = '54fd66c2-b5b5-4dbd-ab89-12a8b6177347'
+    response = @gateway.purchase(@amount, @credit_card, @options)
+
+    assert_success response
+    assert response.params['response'][:Authorized]
+    assert response.params['response'][:OrderId]
+  end
+
+  def test_successful_purchase_with_shipping_data
+    @options[:consumer_price] = '1.01'
+    @options[:consumer_taxes] = '2.01'
+    @options[:consumer_duty] = '1.01'
+
+    @options[:consignee_name] = 'Jane Doe'
+    @options[:consignee_address] = '1670 NW 82ND STR'
+    @options[:consignee_city] = 'Houston'
+    @options[:consignee_country] = 'US'
+
+    response = @gateway.purchase(@amount, @credit_card, @options)
+
+    assert_success response
+    assert response.params['response'][:Authorized]
+    assert response.params['response'][:OrderId]
+  end
+
+  def test_successful_purchase_with_items
```

Comment:
💬 
would be great if we can add some test for the ConsumerTaxes or ConsumerDuty fields

Reply:
New test added for consumer info and consignee, please check.